docs/breaking-changes.md

```diff
@@ -0,0 +1,44 @@
+# Version upgrades and breaking changes
+
+Evolving the Lockbook code base requires some special considerations. 
+
+## Breaking change between core versions
+
+Because we support offline use our clients store and maintain local state. If we update our app, we need to make sure the new version of the app can gracefully interact with the current data on the device. Ultimately we can always nuke the local state and re-sync with the server. We want to avoid doing this where possible for the following reasons:
+
++ We would need to explicitly keep track of offline changes that haven't yet been synced
++ Depending on the user this could take a long time
++ Needlessly places a load our infrastructure
+
+Where possible therefore, where possible we should support old functionality. When this is not possible (a data format changed), a "migration" must be performed. Encoded in rust forever will be instructions on how to step through each version of state. These migrations should not require an internet connection. All core endpoints that interact with state may return `MigrationRequired`, you can also explicitly check the status by calling `get_db_state`.
+
+#### Breaking changes between versions of sled
+
+Sled is the embedded db that lives on all client devices. Sled itself could have breaking changes between versions. The migration process for that is documented [here](https://docs.rs/sled/0.34.4/sled/struct.Db.html#method.export).
+
+## Breaking changes between core and server
+
+Because our clients are native ones installed via arbitrary channels (downloads from our site, package manager, app stores, or compiled from source) our server will explicitly need to keep track of what the minimum version client it can service. We want to be very conservative with incrementing the hard cut off where our server essentially says "I can't service this request until you update your client", being forced to update is generally a bad experience, and it is made worse depending on how you need to update.
```

Comment:
> compiled from source) our ...

Comma before `our`

> ... client", being forced to update ...

New thought, new sentence. Or drop the comma and insert `because`